snapcraft/extensions/_ros2_humble_meta.py

```diff
@@ -0,0 +1,127 @@
+        """Return the ROS 2 Humble related snaps to use to construct the environment."""
+        raise NotImplementedError
+
+    @staticmethod
+    @overrides
+    def is_experimental(base: Optional[str]) -> bool:
+        return True
+
+    @overrides
+    def get_root_snippet(self) -> Dict[str, Any]:
+        root_snippet = super().get_root_snippet()
+        root_snippet["plugs"] = {
+            "ros-humble": {
+                "interface": "content",
+                "content": "ros-humble",
+                "target": "$SNAP/opt/ros/underlay_ws",
+                "default-provider": self.ros2_humble_snaps.content,
+            }
+        }
+        return root_snippet
+
+    @overrides
+    def get_app_snippet(self) -> Dict[str, Any]:
+        app_snippet = super().get_app_snippet()
+        python_paths = app_snippet["environment"]["PYTHONPATH"]
+        new_python_paths = [
+            f"$SNAP/opt/ros/underlay_ws/opt/ros/{self.ROS_DISTRO}/lib/python3.10/site-packages",
+            "$SNAP/opt/ros/underlay_ws/usr/lib/python3/dist-packages",
+        ]
+
+        app_snippet["environment"][
+            "PYTHONPATH"
+        ] = f'{python_paths}:{":".join(new_python_paths)}'
+        app_snippet["environment"]["PATH"] = "$SNAP/opt/ros/underlay_ws/usr/bin:$PATH"
+        app_snippet["environment"][
+            "LD_LIBRARY_PATH"
+        ] = "$SNAP/opt/ros/underlay_ws/usr/lib/$SNAPCRAFT_ARCH_TRIPLET:$LD_LIBRARY_PATH"
+
+        # @todo: for some reason ${VAR:+:$VAR} resulsts in empty substitution
+        # prepend_to_env(
+        #     "PATH", ["$SNAP/opt/ros/underlay_ws/usr/bin"]
+        # )
+        # prepend_to_env(
+        #     "LD_LIBRARY_PATH", ["$SNAP/opt/ros/underlay_ws/usr/lib/$SNAPCRAFT_ARCH_TRIPLET"]
+        # )
+
+        return app_snippet
+
+    @overrides
+    def get_part_snippet(self) -> Dict[str, Any]:
+        part_snippet = super().get_part_snippet()
+        part_snippet["build-snaps"] = [self.ros2_humble_snaps.sdk]
+        part_snippet["colcon-cmake-args"] = [
+            f'-DCMAKE_SYSTEM_PREFIX_PATH="/snap/{self.ros2_humble_snaps.sdk}/current/usr"'
+        ]
+
+        return part_snippet
+
+    @overrides
+    def get_parts_snippet(self) -> Dict[str, Any]:
+        parts_snippet = super().get_parts_snippet()
+        # Very unlikely but it may happen that the snapped application doesn't
+        # even pull those deps. In that case, there is no valid ROS 2 ws to source.
+        # We make sure here that they are staged no matter what.
+        parts_snippet[f"ros2-{self.ROS_DISTRO}/ros2-launch"]["stage-packages"] = [
+            f"ros-{self.ROS_DISTRO}-ros-environment",
+            f"ros-{self.ROS_DISTRO}-ros-workspace",
+            f"ros-{self.ROS_DISTRO}-ament-index-cpp",
+            f"ros-{self.ROS_DISTRO}-ament-index-python",
+        ]
+
+        parts_snippet[f"ros2-{self.ROS_DISTRO}/ros2-launch"]["build-packages"].append(
+            "libpython3.10-dev"
```

Comment:
I'll add a comment. I didn't dig much further but something in the ROS 2 build chain looks for `libpython3.10` during cmake call and can't find it. It can't find it because, while the lib is there, the cmake files to find it aren't. They are shipped with `libpython3.10-dev`.